Moved shared PMU code from pcf50606.h and pcf50633.h (which prevented inclusion
of both at the same time) to pcf506xx.h

- include/linux/pcf50606.h (struct pmu_voltage_rail, enum pmu_event, pmu_cb):
  moved to pcf506xx.h
- include/linux/pcf50633.h (struct pmu_voltage_rail, enum pmu_event, pmu_cb):
  moved to pcf506xx.h

Signed off-by: Werner Almesberger <werner@openmoko.org>

Index: linux-2.6.24-rc7/include/linux/pcf50606.h
===================================================================
--- linux-2.6.24-rc7.orig/include/linux/pcf50606.h
+++ linux-2.6.24-rc7/include/linux/pcf50606.h
@@ -1,6 +1,9 @@
 #ifndef _LINUX_PCF50606_H
 #define _LINUX_PCF50606_H
 
+#include <linux/pcf506xx.h>
+
+
 /* public in-kernel pcf50606 api */
 enum pcf50606_regulator_id {
 	PCF50606_REGULATOR_DCD,
@@ -48,26 +51,6 @@
 extern void
 pcf50606_charge_fast(struct pcf50606_data *pcf, int on);
 
-#define PMU_VRAIL_F_SUSPEND_ON	0x00000001	/* Remains on during suspend */
-#define PMU_VRAIL_F_UNUSED	0x00000002	/* This rail is not used */
-struct pmu_voltage_rail {
-	char *name;
-	unsigned int flags;
-	struct {
-		unsigned int init;
-		unsigned int max;
-	} voltage;
-};
-
-enum pmu_event {
-	PMU_EVT_NONE,
-	PMU_EVT_INSERT,
-	PMU_EVT_REMOVE,
-	__NUM_PMU_EVTS
-};
-
-typedef int pmu_cb(struct device *dev, unsigned int feature,
-		   enum pmu_event event);
 
 #define PCF50606_FEAT_EXTON	0x00000001	/* not yet supported */
 #define PCF50606_FEAT_MBC	0x00000002
Index: linux-2.6.24-rc7/include/linux/pcf50633.h
===================================================================
--- linux-2.6.24-rc7.orig/include/linux/pcf50633.h
+++ linux-2.6.24-rc7/include/linux/pcf50633.h
@@ -1,6 +1,9 @@
 #ifndef _LINUX_PCF50633_H
 #define _LINUX_PCF50633_H
 
+#include <linux/pcf506xx.h>
+
+
 /* public in-kernel pcf50633 api */
 enum pcf50633_regulator_id {
 	PCF50633_REGULATOR_AUTO,
@@ -57,30 +60,6 @@
 extern void
 pcf50633_charge_enable(struct pcf50633_data *pcf, int on);
 
-/* FIXME: sharded with pcf50606 */
-#define PMU_VRAIL_F_SUSPEND_ON	0x00000001	/* Remains on during suspend */
-#define PMU_VRAIL_F_UNUSED	0x00000002	/* This rail is not used */
-struct pmu_voltage_rail {
-	char *name;
-	unsigned int flags;
-	struct {
-		unsigned int init;
-		unsigned int max;
-	} voltage;
-};
-
-enum pmu_event {
-	PMU_EVT_NONE,
-	PMU_EVT_INSERT,
-	PMU_EVT_REMOVE,
-	PMU_EVT_USB_INSERT,
-	PMU_EVT_USB_REMOVE,
-	__NUM_PMU_EVTS
-};
-
-typedef int pmu_cb(struct device *dev, unsigned int feature,
-		   enum pmu_event event);
-
 #define PCF50633_FEAT_EXTON	0x00000001	/* not yet supported */
 #define PCF50633_FEAT_MBC	0x00000002
 #define PCF50633_FEAT_BBC	0x00000004	/* not yet supported */
Index: linux-2.6.24-rc7/include/linux/pcf506xx.h
===================================================================
--- /dev/null
+++ linux-2.6.24-rc7/include/linux/pcf506xx.h
@@ -0,0 +1,31 @@
+#ifndef _LINUX_PCF506XX_H
+#define _LINUX_PCF506XX_H
+
+
+#define PMU_VRAIL_F_SUSPEND_ON	0x00000001	/* Remains on during suspend */
+#define PMU_VRAIL_F_UNUSED	0x00000002	/* This rail is not used */
+struct pmu_voltage_rail {
+	char *name;
+	unsigned int flags;
+	struct {
+		unsigned int init;
+		unsigned int max;
+	} voltage;
+};
+
+enum pmu_event {
+	PMU_EVT_NONE,
+	PMU_EVT_INSERT,
+	PMU_EVT_REMOVE,
+#ifdef CONFIG_SENSORS_PCF50633
+	PMU_EVT_USB_INSERT,
+	PMU_EVT_USB_REMOVE,
+#endif
+	__NUM_PMU_EVTS
+};
+
+typedef int pmu_cb(struct device *dev, unsigned int feature,
+		   enum pmu_event event);
+
+
+#endif /* !_LINUX_PCF506XX_H */
